NitroxClient/MonoBehaviours/Multiplayer.cs

```diff
@@ -85,112 +48,57 @@ public void ProcessPackets()
 
             foreach (Packet packet in packets)
             {
-                if (PacketProcessorsByType.ContainsKey(packet.GetType()))
+                try
                 {
-                    try
-                    {
-                        PacketProcessor processor = PacketProcessorsByType[packet.GetType()];
-                        processor.ProcessPacket(packet, null);
-                    }
-                    catch (Exception ex)
-                    {
-                        Log.Error("Error processing packet: " + packet, ex);
-                    }
+                    Type clientPacketProcessorType = typeof(ClientPacketProcessor<>);
```

Comment:
Don't we want to preprocess this once like in `MultiplayerClient.GeneratePacketProcessorMap`? I have not profiled it but I'd not be surprised if this is slow.

Reply:
Not slow as far as I can tell. I've done this exact same thing when routing messages received on a REST endpoint into the appropriate DDD service.

Reply:
As discussed yesterday, it's probably fine, though I do like to profile the difference between direct lookup from a map and reconstructing the generic type for every packet. Every CPU cycle we can spare is one; the game itself is heavy as is already.

Reply:
Been doing some research. The use of `typeof` incurs far less of a performance cost at run-time than `GetType` because it is a static type lookup that is resolved once at compile-time.

Calling `MakeGenericType` does take longer than grabbing the resolved type from a dictionary, but it only takes about 10 - 20 ticks in total from what I can gather from research. It feels like excessive optimization to be overly concerned about it.

At any rate - simply caching the generic type reference after it is initially resolved would be the ideal solution in my mind. Scanning the entire assembly can get expensive and I'd hate for the mod to put too much of a burden on the initial load than is necessary.

Seems like something that can be easily addressed if/when it becomes a problem.